### ISSUE TITLE
Revisit cache key computation for copy steps

### ISSUE THREAD

**Issue by dlorenc** [GoogleContainerTools/kaniko#574]
Right now this uses the contents of the files being copied. Instead, we could use the file paths and the base image digest.

This would allow us to calculate the cache key from a COPY line and the base image digest without having to fetch files.

**Comment by priyawadhwa** [comment]
I initially included the contents so that if a file was modified the cache key would be different, but maybe there's another way to do it!

**Comment by dlorenc** [comment]
Yeah the contents work, but so should the filepath + image digest. I don't mean base image here, I mean something like:

```
FROM foo as stage1
RUN bar

FROM baz as stage2
COPY --from=1 /bat/baz /foo
```

the key for the copy instruction can be the digest of "stage1", plus the filepaths in the src. If any of those have changed, stage1 will have a different digest.

Does that make sense?

**Comment by priyawadhwa** [comment]
ohhh gotcha, thanks for explaining!
